api/__tests__/profile.spec.ts

```diff
@@ -105,6 +155,96 @@ describe('Profile event handlers', () => {
 
     // @ts-ignore
     await expect(fetchProjectProfile(req, ex.res)).rejects.toThrowErrorMatchingSnapshot();
+
+    expect(pmock.query.mock.calls).toMatchSnapshot();
     expect(ex.responseData).toBeUndefined();
   });
+
+  it('A project is updated', async () => {
+    const body = JSON.parse(JSON.stringify(insertProfile));
+    const aBody = {
+      id: 9,
+      ...body,
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
+    };
+    const req = {
+      params: { profileId: 1 },
+      body: aBody,
+    }
+
+    pmock.query.mockReturnValue({ rows: [aBody] });
+
+    // @ts-ignore
+    await updateProjectProfile(req, ex.res);
+
+    expect(pmock.query.mock.calls).toMatchSnapshot();
+    expect(ex.res.statusCode).toMatchSnapshot();
+    expect(ex.responseData).toMatchSnapshot();
+    expect(ex.res.status).toBeCalled();
+    expect(ex.res.json).toBeCalled();
+  });
+
+  it('A project fails to update', async () => {
+    const body = JSON.parse(JSON.stringify(insertProfile));
+    const aBody = {
+      id: 9,
+      ...body,
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
+    };
+    const req = {
+      params: { profileId: 1 },
+      body: aBody,
+    }
+
+    pmock.query.mockImplementation(() => { throw new Error() });
+
+    // @ts-ignore
+    await expect(updateProjectProfile(req, ex.res)).rejects.toThrow();
+
+    expect(ex.res.status).not.toBeCalled();
+    expect(ex.res.json).not.toBeCalled();
+  });
+
+  it('A project is archived', async () => {
+    const aBody = {
+      id: 9,
+      ...insertProfile,
+      createdAt: Date.now(),
```

Comment:
unsure why Date.now has to be mocked

Reply:
The result is eventually recorded by a snapshot. If it gives different times the snapshot changes with each run.